# Experiment 026262740cfcd15e14f11ab6637c23adcd86cb714cfd8cac03d909b9cbfc947c
# Base df58388dfc9d3e2395e88258d4170e2459e7e248
diff --git a/src/scr_cred.cpp b/src/scr_cred.cpp
index 72d522fe1..782a4f899 100644
--- a/src/scr_cred.cpp
+++ b/src/scr_cred.cpp
@@ -26,7 +26,7 @@ extern int DAT_0086b2a0;
 extern unsigned char DAT_0086b2a4;
 extern tagPALETTEENTRY DAT_0086b2a8;
 extern float DAT_0086b2ac;
-extern tagPALETTEENTRY DAT_0086b2b0[256];
+extern tagPALETTEENTRY DAT_0086b2b0[];
 extern tagPALETTEENTRY DAT_0086b6b0[256];
 extern TDrawArea* DAT_0086bab0;
 
@@ -513,7 +513,7 @@ void RGE_fade_palette_start(TDrawArea* draw_area, tagPALETTEENTRY fade_color,
             target->peFlags = source->peFlags;
             ++target;
             ++source;
-        } while (target < DAT_0086b2b0 + 256);
+        } while ((int)target < (int)DAT_0086b6b0);
     }
 
     DAT_0086b294 = debug_timeGetTime("C:\\msdev\\work\\age1_x1\\scr_cred.cpp", 0x2B5);
